File: codeforces/ac/0/039/D.hs
main = do interact (gao . read)
gao n = if a!!y /= m then "NO" else unlines $ "YES": map show c where
	a = let f i j = i: f j (13 * j - 12 * i) in f 2 13
	x = last $ takeWhile ((==0) . mod n . (12^)) [0 ..]
	m = div n (12 ^ x)
	y = head $ dropWhile ((<m) . (a!!)) [0 ..]
	z = 2 * x + y
	b = [a!!(z - 2 * i) * 12 ^ i | i <- reverse [0 .. div z 2], i /= x]
	c = 1: toInteger z + 1: toInteger (length b): b
{-
# 	When 	Who 	Problem 	Lang 	Verdict 	Time 	Memory
235808 	Jan 1, 2011 6:14:13 PM 	watashi 	D - Interesting Sequence 	Haskell 	Accepted 	50 ms 	1548 KB 
-}

File: codeforces/ac/0/039/D.rb
A = a = gets.to_i

x = 0
while a % 12 == 0
	a /= 12
	x += 1
end

b = [2, 13]
999.times{b.push 13 * b[-1] - 12 * b[-2]}	# 300.times => RE
y = 0
y += 1 while b[y] < a

if b[y] != a
	puts :NO
else
	z = 2 * x + y
	c = []
	(z / 2).downto(0){|i| c << b[z - 2 * i] * 12 ** i if i != x}
	puts :YES
	p 1, z + 1, c.size, *c
end

__END__
# 	When 	Who 	Problem 	Lang 	Verdict 	Time 	Memory
235802 	Jan 1, 2011 5:48:26 PM 	watashi 	D - Interesting Sequence 	Ruby 	Accepted 	90 ms 	3804 KB

# a b (13b-12a) 12a => no matter which method, both get 12*12b !!
# so we can always put all 13b[-1]-12b[-2] at the begining of sequence !!
